src/gov/nist/javax/sdp/parser/OriginFieldParser.java

```diff
@@ -59,11 +59,11 @@ public OriginField originField() throws ParseException {
             Token sessionId = lexer.getNextToken();
             // guard against very long session IDs
             String sessId = sessionId.getTokenValue();
-            if (sessId.length() > 18)
-                sessId = sessId.substring(sessId.length() - 18);
             try {
                 originField.setSessId(Long.parseLong(sessId));
             } catch (NumberFormatException ex) {
+                if (sessId.length() > 18)
```

Comment:
The problem is, that Long.MAX_VALUE is 9223372036854775807 and therefore not all numbers with 19 digits are valid. My suggestion is to try the parseLong() and in case of an Exception shorten it to 18 digits to be sure (as it was before). Somewhere else the string value here will be parsed again if no "sessId" has been set here and there you will have the same exception when shortening to 19 instead of 18.